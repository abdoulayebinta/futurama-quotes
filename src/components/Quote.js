import React from 'react';

const Quote = ({ quote, onQuoteSelect }) => {
    
    const imageUrl = quote.image;

    return (
        <li onClick={() => onQuoteSelect(quote)} className="list-group-item">
            <div className="media">
                <img className="mr-3 mini-quote-img" src={imageUrl} />
                <div className="media-body">
                    <h5 className="mt-0">{quote.character}</h5>
                    <p>{quote.quote.substring(0, 20)}...</p>
                </div>
            </div>
        </li>
    );
};

export default Quote;
