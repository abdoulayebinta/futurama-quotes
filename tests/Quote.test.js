import React from 'react';
import renderer from 'react-test-renderer';
import Quote from '../src/components/Quote';

const testQuote = {
    'character': 'Fry', 
    'quote': "I haven't had time off since I was 21 through 24.", 
    'image': 'Fry.png'
}

describe('<Quote />', () => {
  test('Should break if no quote passed', () => {
    const component = renderer.create(<Quote quote={testQuote} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should correctly rendered quote', () => {
    const component = renderer.create(<Quote quote={testQuote} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should render selected quote', () => {
    const component = renderer.create(<Quote quote={testQuote} onQuoteSelect={testQuote} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
