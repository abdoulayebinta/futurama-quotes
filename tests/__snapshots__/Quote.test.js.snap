// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Quote /> Should break if no quote passed 1`] = `
<li
  className="list-group-item"
  onClick={[Function]}
>
  <div
    className="media"
  >
    <img
      alt="Fry"
      className="mr-3 mini-quote-img"
      src="Fry.png"
    />
    <div
      className="media-body"
    >
      <h5
        className="mt-0"
      >
        Fry
      </h5>
      <p>
        I haven't had time o
        ...
      </p>
    </div>
  </div>
</li>
`;

exports[`<Quote /> Should correctly rendered quote 1`] = `
<li
  className="list-group-item"
  onClick={[Function]}
>
  <div
    className="media"
  >
    <img
      alt="Fry"
      className="mr-3 mini-quote-img"
      src="Fry.png"
    />
    <div
      className="media-body"
    >
      <h5
        className="mt-0"
      >
        Fry
      </h5>
      <p>
        I haven't had time o
        ...
      </p>
    </div>
  </div>
</li>
`;

exports[`<Quote /> Should not break if no quote passed 1`] = `
<li
  className="list-group-item"
  onClick={[Function]}
>
  <div
    className="media"
  >
    <img
      alt="Fry"
      className="mr-3 mini-quote-img"
      src="Fry.png"
    />
    <div
      className="media-body"
    >
      <h5
        className="mt-0"
      >
        Fry
      </h5>
      <p>
        I haven't had time o
        ...
      </p>
    </div>
  </div>
</li>
`;

exports[`<Quote /> Should render selected quote 1`] = `
<li
  className="list-group-item"
  onClick={[Function]}
>
  <div
    className="media"
  >
    <img
      alt="Fry"
      className="mr-3 mini-quote-img"
      src="Fry.png"
    />
    <div
      className="media-body"
    >
      <h5
        className="mt-0"
      >
        Fry
      </h5>
      <p>
        I haven't had time o
        ...
      </p>
    </div>
  </div>
</li>
`;
